### PR TITLE
Dependencies Update, New Tray Behavior, and some random improvements :)

### PR DESCRIPTION
**Pull request by bagusnl**
[Build-Canary](https://github.com/CollapseLauncher/Collapse/actions/workflows/build.yml)
In this PR:
- **[IMP]** Added MSBuild optimization for those using CoW based FS for their Git repository (eg. ReFS)
- **[NEW]** Add Git short SHA where the build is compiled on console
- **[IMP]** Update CommandLine to beta4 (bugfixes, NativeAOT Compile and ILTrim), include all adjustments necessary
- **[VULN]** Update HTTP and Regex NuGet (CVE-2019-0820)
- **[NEW]** New TrayIcon using [H.NotifyIcon](https://github.com/HavenDV/H.NotifyIcon)
- **[NEW]** Add new start argument to start Collapse on system tray

TODO (ON NEXT PR):
- Make it possible to run Collapse at startup (bonus point if user can choose between start normally and start on tray)
- Make it possible to exit to tray (hijacks usual close button to be configurable)
- #170 ???
- Code checks

Template:
**[FIX]** 
**[IMP]** 
**[VUL]** 
**[NEW]**

### CODE REVIEW

File: CollapseLauncher/CollapseLauncher.csproj [hunk 1]
```diff
@@ -37,21 +38,31 @@
...
 	<ItemGroup>
 		<PackageReference Include="Clowd.Squirrel" Version="2.9.42" />
+		<PackageReference Include="CommunityToolkit.Mvvm" Version="8.2.1" />
 		<PackageReference Include="CommunityToolkit.WinUI.UI.Controls" Version="7.1.2" />
+		<PackageReference Include="GitInfo" Version="3.3.3">
+		  <PrivateAssets>all</PrivateAssets>
+		  <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+		</PackageReference>
+		<PackageReference Include="H.NotifyIcon.WinUI" Version="2.0.115" />
+		<PackageReference Include="Microsoft.Windows.CsWinRT" Version="2.0.3" />
```
Comment (bagusnl) [inline CollapseLauncher/CollapseLauncher.csproj:58]: Unused

**Review by neon-nyan — APPROVED** [review]
Tested! LGTM

**Review by Cryotechnic — APPROVED** [review]
LGTM, have some commits OTW which will fix some of the issues discussed internally

**Review by muscularcandy67 — APPROVED** [review]
LGTM too!

**Pull request merged**